txp/Pos/Txp/Toil/Utxo/Functions.hs

```diff
@@ -250,5 +266,9 @@ rollbackTxUtxo
 rollbackTxUtxo (txAux, undo) = do
     let tx@UnsafeTx {..} = taTx txAux
     let txid = hash tx
-    mapM_ utxoDel $ take (length _txOutputs) $ map (TxIn txid) [0..]
-    mapM_ (uncurry utxoPut) $ NE.zip _txInputs undo
+    mapM_ utxoDel $ take (length _txOutputs) $ map (TxInUtxo txid) [0..]
+    mapM_ (uncurry utxoPut) $ concatMap knownInputAndUndo $ toList $ NE.zip _txInputs undo
+  where
+    knownInputAndUndo (_, Nothing)         = []
+    knownInputAndUndo (TxInUnknown _ _, _) = []
+    knownInputAndUndo (inp, Just u)        = [(inp, u)]
```

Comment:
Well, it would be good to at least `logWarning` then, but it looks complicated, because we don't have `WithLogger` here (unlike in update system), so ok, let's leave it how it is.